### PR TITLE
feat(usersManagement): add users management for admin and super admin

### PR DESCRIPTION
**Pull request by ItayBenMoshe**
[433](https://dev.azure.com/spectrumFactory/CoronaI/_backlogs/backlog/CoronaI%20Team/Features/?workitem=433)
For now admin and superAdmin see the same table.
At the future SuperAdmin will be able to update investigator data such as 'investigations Group', 'source Organization'

### CODE REVIEW

File: client/src/components/App/AppToolbar/AppToolbar.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { Typography } from '@material-ui/core';
 import Tooltip from '@material-ui/core/Tooltip';
 
 import useAppToolbar from './useAppToolbar';
-import IsActiveToggle from './IsActiveToggle/IsActiveToggle';
+import IsActiveToggle from 'commons/IsActiveToggle/IsActiveToggle';
```
Comment (MayaSasson) [inline client/src/components/App/AppToolbar/AppToolbar.tsx:7]: Imports order

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
```
Comment (MayaSasson) [inline client/src/components/App/Content/UsersManagement/UsersManagement.tsx:1]: Isn’t this component a bit empty? Are you going to add anything else besides the table?

Comment (ItayBenMoshe) [reply client/src/components/App/Content/UsersManagement/UsersManagement.tsx:1]: Yes you right, Naaman already told me about it yesterday

File: client/src/components/App/Content/UsersManagement/UsersManagementTable/UsersManagementTable.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                    <TableBody>
+                        {
+                            users.map((user: any) => ( 
+                                <TableRow>
+                                    { 
+                                        Object.keys(UsersManagementTableHeaders).map(cellName => (
+                                            <TableCell>
+                                                {
+                                                    getTableCell(user, cellName)
+
```
Comment (MayaSasson) [inline client/src/components/App/Content/UsersManagement/UsersManagementTable/UsersManagementTable.tsx:76]: Delete row

File: client/src/components/App/Content/UsersManagement/UsersManagementTable/useUsersManagementTable.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import { Service, Severity } from 'models/Logger'
+import StoreStateType from 'redux/storeStateType'
+import axios from 'Utils/axios'
+
+const useUsersManagementTable = ({ page, rowsPerPage}: useUsersManagementTableInCome) : useUsersManagementTableOutCome => {
+
+    const userId = useSelector<StoreStateType, string>(state => state.user.id);
+    const epidemiologyNumber = useSelector<StoreStateType, number>(state => state.investigation.epidemiologyNumber);
+
+    const [users, setUsers] = useState<any>([]);
```
Comment (MayaSasson) [inline client/src/components/App/Content/UsersManagement/UsersManagementTable/useUsersManagementTable.ts:15]: Why any?

File: client/src/components/App/Content/UsersManagement/UsersManagementTable/useUsersManagementTable.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            severity: Severity.LOW,
+            workflow: 'Fetching users',
+            step: 'launching users request',
+            user: userId,
+            investigation: epidemiologyNumber
+        })
+        axios.post('/users', { page: { number: page, size: rowsPerPage } })
+            .then(result => {
+                result?.data && setUsers(result.data?.users);
+                result?.data && setTotalCount(result.data?.totalCount);
```
Comment (MayaSasson) [inline client/src/components/App/Content/UsersManagement/UsersManagementTable/useUsersManagementTable.ts:30]: Why do you do the same check twice?

File: server/src/ClientToDBAPI/UsersRoute/mainRoute.ts [hunk 6]
```diff
@@ -171,6 +172,38 @@ usersRoute.post('/changeInvestigator', adminMiddleWare, (request: Request, respo
...
+    logger.info({
+        service: Service.SERVER,
+        severity: Severity.LOW,
+        workflow: 'Getting user types',
+        step: 'querying the graphql API',
+        user: response.locals.user.id
+    });
+    graphqlRequest(GET_ALL_USER_TYPES, response.locals)
+        .then((result: GetAllUserTypesResponse) => {
+            if (result && result.data && result.data.allUserTypes) {
```
Comment (MayaSasson) [inline server/src/ClientToDBAPI/UsersRoute/mainRoute.ts:185]: You can use the ?. syntax

Comment (MayaSasson) [reply server/src/ClientToDBAPI/UsersRoute/mainRoute.ts:185]: In addition, you need to add log for the else clause and send response with error message

File: client/src/components/App/AppToolbar/AppToolbar.tsx (hunk 1 above)
Comment (ItayBenMoshe) [reply client/src/components/App/AppToolbar/AppToolbar.tsx:6]: fixed

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx (hunk 2 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/UsersManagement/UsersManagement.tsx]: fixed

File: client/src/components/App/Content/UsersManagement/UsersManagementTable/UsersManagementTable.tsx (hunk 3 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/UsersManagement/UsersManagementTable/UsersManagementTable.tsx]: fixed

File: client/src/components/App/Content/UsersManagement/UsersManagementTable/useUsersManagementTable.ts (hunk 4 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/UsersManagement/UsersManagementTable/useUsersManagementTable.ts]: fixed

File: client/src/components/App/Content/UsersManagement/UsersManagementTable/useUsersManagementTable.ts (hunk 5 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/UsersManagement/UsersManagementTable/useUsersManagementTable.ts]: fixed

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+import IsActiveToggle from 'commons/IsActiveToggle/IsActiveToggle'
+
+import { UsersManagementTableHeaders, UsersManagementTableHeadersNames } from './UsersManagementTableHeaders'
+import useStyles from './UsersManagementStyles'
+import useUsersManagementTable from './useUsersManagement';
+
+const rowsPerPage: number = 7;
+
+const UsersManagementTable: React.FC = () => {
```
Comment (NaamanTsur) [inline client/src/components/App/Content/UsersManagement/UsersManagement.tsx:15]: The component name should be the same as the file name.

File: client/src/components/App/Content/UsersManagement/UsersManagementStyles.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import { Theme } from '@material-ui/core';
+import { makeStyles } from '@material-ui/styles';
+
+const useStyles = makeStyles((theme: Theme) => ({
+    content: {
+        height: '84vh',
+        backgroundColor: '#F3F6FB',
```
Comment (NaamanTsur) [inline client/src/components/App/Content/UsersManagement/UsersManagementStyles.ts:7]: See whether you can get the background color from the theme object

**Review by NaamanTsur — CHANGES_REQUESTED** [review]

File: server/src/ClientToDBAPI/UsersRoute/mainRoute.ts (hunk 6 above)
Comment (ItayBenMoshe) [reply server/src/ClientToDBAPI/UsersRoute/mainRoute.ts]: fixed

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx (hunk 7 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/UsersManagement/UsersManagement.tsx]: fixed

File: client/src/components/App/Content/UsersManagement/UsersManagementStyles.ts (hunk 8 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/UsersManagement/UsersManagementStyles.ts]: fixed

**Review by NaamanTsur — APPROVED** [review]

**Review by MayaSasson — APPROVED** [review]

**Pull request merged**